GUIDELINES.md

```diff
@@ -0,0 +1,41 @@
+# Instrumentations Implementation Guide
+
+This document captures general guidelines for implementing instrumentations in NodeJS and browser.
+
+## Types
+
+### Public Types
+
+Public types are meant to be consumed by instrumentation users (OpenTelemetry distribution packages or end users implementing OpenTelemetry in their services). These are mostely instrumentation specific config interface (extending `InstrumentationConfig`) and the transitive types used in the config.
```

Comment:
Should we mention somewhere that re-exporting (and therefore depending on) types from the instrumented library or it's `@types` package is not allowed/should be avoided?

Reply:
Yes, we should.
I wanted to keep this PR as short as possible and only address one thing.
After it is merged, I'll open follow-up PRs for other issues such as the one you mentioned.